doc/tutorials/debug.rst

```diff
@@ -0,0 +1,371 @@
+number, we will know that pCPU active bitmap is 0x000000000000000f and
+pCPU number is 0x0000000000000004.
+
+.. figure:: images/debug_image12.png
+   :align: center
+
+   dumpmem information
+
+.. figure:: images/debug_image11.png
+   :align: center
+
+   acrn map information
+
+dump_ioapic
+===========
+
+"dump_ioapic" provides IOAPIC information and we can get IRQ number,
+IRQ vector number, etc.
+
+.. figure:: images/debug_image14.png
+   :align: center
+
+   dump_ioapic information
+
+pt
+==
+
+"pt" provides pass-through detailed information, such as the virtual
+machine number, interrupt type, interrupt request, interrupt vector,
+trigger mode, etc.
+
+.. figure:: images/debug_image13.png
+   :align: center
+
+   pt information
+
+int
+===
+
+"int" provides interrupt information on all CPUs and their corresponding
+interrupt vector.
+
+.. figure:: images/debug_image17.png
+   :align: center
+
+   int information
+
+cpuid
+=====
+
+"cupid cpu_num" provides the highest EAX call parameter (leaf)
+supported by the CPU in the EAX register and the values of EBX, ECX, and EDX
+registers.
+
+.. figure:: images/debug_image15.png
+   :align: center
+
+   cpuid information
+
+rdmsr/wrmsr
+===========
+
+We can read and write model specific register (MSR) to get and set register
+values through ``rdmsr/wrmsr register_address``.
+
+In the following example, we can get IA32_APIC_BASE value through
+the command::
+
+   rdmsr 1b
+
+and see that 1B (Hexadecimal) is the IA32_APIC_BASE MSR address.
+
+.. figure:: images/debug_image16.png
+   :align: center
+
+   IA32_APIC_BASE register information
+
+.. figure:: images/debug_image18.png
+   :align: center
+
+   rdmsr information
+
+An example
+==========
+
+As an example, we'll show how to obtain the interrupts of a pass-through USB device.
+
+First, we can get the USB controller BDF number (0:15.0) through the
+following command::
+
+   lspci | grep "USB controller"
+
+
+.. figure:: images/debug_image19.png
+   :align: center
+
+   USB controller BDF information
+
+Second, we use the ``pt`` command and use this BDF number
+to find the interrupt vector (VEC) "0x31".
+
+.. figure:: images/debug_image20.png
+   :align: center
+
+   USB controller interrupt information
+
+Finally we use the ``int`` command and use this interrupt vector
+to find the interrupt number (909) on CPU3.
+
+.. figure:: images/debug_image21.png
+   :align: center
+
+   USB controller interrupt number information
+
+ACRN Log
+********
+
+ACRN log provides console log and acrnlog for a user to analyze.
+We can use console log to debug directly, while acrnlog is a userland tool
+used to capture a ACRN hypervisor log.
+
+console log/acrn log
+====================
+
+We can set console log level and acrnlog level through::
+
+   loglevel [console_loglevel] [mem_loglevel]
+
+For example, we add the following code into function ``shell_cmd_help`` of
+the source file ``acrn-hypervisor/hypervisor/debug/shell.c``..
```

Comment:
I don't understand what you are trying to achieve/demonstrate it. Can you clarify?